hydromt/models/root.py

```diff
@@ -58,6 +86,30 @@ def is_reading_mode(self) -> bool:
         """Test whether we are in reading mode or not."""
         return self._mode.is_reading_mode()
 
+    # old_root is so we can copy over the old log file if needed.
+    def _create_loggers(self, old_path: Optional[Path] = None, overwrite: bool = False):
+        if not exists(self._path):
+            mkdir(self._path)
```

Comment:
I always use os.makedirs which recursively creates folders, rather than just a subfolder.
It also has an argument `exist_ok` which avoids the if statement

```suggestion
        os.makedirs(self._path, exist_ok=True)
```

Reply:
We agreed that the root should only create the root itself not any of the subfolders, so it is the same in this case

Reply:
We do want to be able to create a new root `"/c/models/sfincs/delft"` in a folder `"/c/models/"`.  With `os.mkdir` you get a FileNotFoundError because `"/c/models/sfincs"` does not yet exist. `os.makedirs` solves this as it recursively creates the requested root folder.

With subfolders I meant that in v0.9 the `set_root` creates several subfolderes within the root based on the `Model._FOLDERS` class property, e.g. `"<root>/geoms"` in WflowModel to save all geoms. The responsibility to create the geoms subfolder should be moved to the geoms write method.

Reply:
fair enough.